src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/LinuxNetworkInterface.cs

```diff
@@ -16,15 +16,48 @@ internal class LinuxNetworkInterface : UnixNetworkInterface
         private readonly OperationalStatus _operationalStatus;
         private readonly bool? _supportsMulticast;
         private readonly long? _speed;
+        internal readonly int _mtu;
         private readonly LinuxIPInterfaceProperties _ipProperties;
 
-        internal LinuxNetworkInterface(string name, int index) : base(name)
+        internal class LinuxNetworkInterfaceSystemProperties
+        {
+            internal string[] IPv4Routes;
+            internal string[] IPv6Routes;
+            internal string DnsSuffix;
+            internal IPAddressCollection DnsAddresses;
+
+            internal LinuxNetworkInterfaceSystemProperties()
+            {
+                if (File.Exists(NetworkFiles.Ipv4RouteFile))
+                {
+                    IPv4Routes = File.ReadAllLines(NetworkFiles.Ipv4RouteFile);
+                }
+
+                if (File.Exists(NetworkFiles.Ipv6RouteFile))
+                {
+                    IPv6Routes = File.ReadAllLines(NetworkFiles.Ipv6RouteFile);
+                }
+
+                try
+                {
+                    string resolverConfig = File.ReadAllText(NetworkFiles.EtcResolvConfFile);
```

Comment:
In addition to the try/catch, should we also check whether the File.Exists as we do above, to avoid the exception in the failure case?  Or is that very rare?

Reply:
I think that would be pretty rare. Without it system would depend only on entries in /etc/hosts. The question is if we should throw in such case or if we should return empty collection. It seems like throwing PNSP for valid configuration is not ideal IMHO.